server/region_handler.go

```diff
@@ -647,17 +653,82 @@ func (ir RegionFrameRange) lastTableID() int64 {
 	return ir.last.TableID
 }
 
+// parseQuery is used to parse query string in URL, due to golang http package can not distinguish
+// query like "?a=" and "?a". We rewrite it to separate these two queries. e.g.
+// "?a=" which means that a is an empty string "";
+// "?a"  which means that a is null.
+func parseQuery(query string, m url.Values) error {
+	var err error
+	for query != "" {
+		key := query
+		if i := strings.IndexAny(key, "&;"); i >= 0 {
+			key, query = key[:i], key[i+1:]
+		} else {
+			query = ""
+		}
+		if key == "" {
+			continue
+		}
+		if i := strings.Index(key, "="); i >= 0 {
+			value := ""
+			var err1 error
+			key, value = key[:i], key[i+1:]
+			key, err1 = url.QueryUnescape(key)
+			if err1 != nil {
+				if err == nil {
+					err = err1
```

Comment:
Could we return err here?

Reply:
OK. I just follow the style of url package. Fixed.